### PR TITLE
Minor bugfix simulation 

### LINKED ISSUE #69: TypeError at /api/cases/counterfactual/daily/absolute/
Issue with API call for /api/cases/counterfactual/daily/absolute/

```
Internal Server Error: /api/cases/counterfactual/daily/absolute/
backend_1   | Traceback (most recent call last):
backend_1   |   File "/root/.cache/pypoetry/virtualenvs/counterfactualcovid19-vsmFjuTQ-py3.9/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
backend_1   |     response = get_response(request)
backend_1   |   File "/root/.cache/pypoetry/virtualenvs/counterfactualcovid19-vsmFjuTQ-py3.9/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
backend_1   |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
backend_1   |   File "/root/.cache/pypoetry/virtualenvs/counterfactualcovid19-vsmFjuTQ-py3.9/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
backend_1   |     return view_func(*args, **kwargs)
backend_1   |   File "/root/.cache/pypoetry/virtualenvs/counterfactualcovid19-vsmFjuTQ-py3.9/lib/python3.9/site-packages/rest_framework/viewsets.py", line 125, in view
backend_1   |     return self.dispatch(request, *args, **kwargs)
backend_1   |   File "/root/.cache/pypoetry/virtualenvs/counterfactualcovid19-vsmFjuTQ-py3.9/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
backend_1   |     response = self.handle_exception(exc)
backend_1   |   File "/root/.cache/pypoetry/virtualenvs/counterfactualcovid19-vsmFjuTQ-py3.9/lib/python3.9/site-packages/rest_framework/views.py", line 469, in handle_exception
backend_1   |     self.raise_uncaught_exception(exc)
backend_1   |   File "/root/.cache/pypoetry/virtualenvs/counterfactualcovid19-vsmFjuTQ-py3.9/lib/python3.9/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
backend_1   |     raise exc
backend_1   |   File "/root/.cache/pypoetry/virtualenvs/counterfactualcovid19-vsmFjuTQ-py3.9/lib/python3.9/site-packages/rest_framework/views.py", line 506, in dispatch
backend_1   |     response = handler(request, *args, **kwargs)
backend_1   |   File "/app/backend/cases/views.py", line 46, in list
backend_1   |     return Response(serializer.data)  # pylint: disable=no-member
backend_1   |   File "/root/.cache/pypoetry/virtualenvs/counterfactualcovid19-vsmFjuTQ-py3.9/lib/python3.9/site-packages/rest_framework/serializers.py", line 745, in data
backend_1   |     ret = super().data
backend_1   |   File "/root/.cache/pypoetry/virtualenvs/counterfactualcovid19-vsmFjuTQ-py3.9/lib/python3.9/site-packages/rest_framework/serializers.py", line 246, in data
backend_1   |     self._data = self.to_representation(self.instance)
backend_1   |   File "/root/.cache/pypoetry/virtualenvs/counterfactualcovid19-vsmFjuTQ-py3.9/lib/python3.9/site-packages/rest_framework/serializers.py", line 663, in to_representation
backend_1   |     return [
backend_1   | TypeError: 'type' object is not iterable
```

### LINKED ISSUE #70: Simulation initial number of cases offset applied too early on the time series
Example, the Norway model. The ideal would be that cases are 0 up to the start day of the simulation, and then the offset gets applied. 

Screenshot 2021-04-06 at 14 56 50

### PR DESCRIPTION
**Pull request by crangelsmith**
- [x] Fixes #70 
- [x] Removes old simulation scripts
- [x] Fixes #69
- [x] Fix broken test

### CODE REVIEW

**Comment by crangelsmith** [comment]
Hi @jemrobinson, have you seen the error that is making the [CI fail](https://github.com/alan-turing-institute/CounterfactualCovid19/pull/71/checks?check_run_id=2286483986) for this PR?

The tests run ok locally (in docker) and I don't see anything obvious changing in this PR wrt branch #65 that would create the problem.

**Comment by jemrobinson** [comment]
It doesn't look like a problem with anything that you've changed, but Django imports different settings for running natively and in Docker. I'll take a look later.

OK, looks like the default Django behaviour has changed (https://docs.djangoproject.com/en/3.1/ref/applications/).

Should be fixed in d78f689 and d14fb43 on this branch.

File: backend/counterfactual/simulation.py [hunk 1]
```diff
@@ -256,7 +256,9 @@ def simulate_dataframes(
 
         # Add cumulative cases that occurred before the simulation start date
         df_counterfactuals.append(
-            add_cumulative_sum(df_counterfactual_country, initial_cum_cases)
+            add_cumulative_sum(
+                df_counterfactual_country, initial_cum_cases, simulation_start_date
+            )
```
Comment (jemrobinson) [inline backend/counterfactual/simulation.py:261]: I think that the first date in `df_counterfactual_country` will always be `simulation_start_date` so I'm not sure that this change is needed. Maybe I'm missing something though?

Comment (crangelsmith) [reply backend/counterfactual/simulation.py:261]: Not really, df_counterfactual_country has the same date range that the real data dataframes (starting from January), but the modelling only happens in a subset of those ranges. Having different date ranges breaks the front end histogram that expects the data to have equal shapes. This shouldn't be an issue in most cases, but in some special countries like Norway not adding this step makes the histogram look like in issue #70

Comment (jemrobinson) [reply backend/counterfactual/simulation.py:261]: So `df_counterfactual_country` comes from calling `simulate_single_country` with arguments of `(simulation_start_date, simulation_end_date)`. On line 320, you can see that the `simulation_dates` date range being used in this function runs from `simulation_start_date` to `simulation_end_date`. This is a change that I made when I was pulling these functions out into a separate function, so if it's wrong then feel free to revert to the previous behaviour.

Comment (crangelsmith) [reply backend/counterfactual/simulation.py:261]: Yes, but in line 381 the resulting simulation dataframe gets outer-merged with the real case data which extends the date range, and all non common dates get filled with 0 (which would create the problem in Norway).

I don't think is wrong, in fact it is the same behaviour we have in the previous version.

**Comment by crangelsmith** [comment]
Anyway, @jemrobinson if you think there is a better solution to this kind of hack-y fix to issue #70, lets talk about it today in our meeting. If not I think this PR is RTM.

**Review by jemrobinson — APPROVED** [review]
LGTM. Merge whenever you're ready @crangelsmith

**Pull request merged**